### ISSUE TITLE
[SPA] Redirect to external URL from asyncData replaces current history entry

### ISSUE THREAD

**Issue by rchl** [nuxt/nuxt.js#5032]
### Version

[v2.4.3](https://github.com/nuxt.js/releases/tag/v2.4.3)

### Reproduction link

[https://codesandbox.io/s/8yl9852pv9](https://codesandbox.io/s/8yl9852pv9)

### Steps to reproduce

1. Create `pages/fail.vue` with:
```
<template>
  <div>
    FAIL
  </div>
</template>
```

2. Create `pages/redirect.vue` with:

```
<script>
export default {
  asyncData({ redirect }) {
    redirect("https://elg.no");
  }
};
</script>
```

3. Create `pages/index.vue` with:

```
<template>
  <nuxt-lint to="/redirect">Navigate to external page</nuxt-link>
</template>
```

4. Navigate to /fail
5. Navigate to /
6. Click the link
7. Navigate back in history

### What is expected ?

Return to the index page where link was clicked from.

### What is actually happening?

Navigates to /fail page

### Additional comments?

What happens is, when navigating in SPA mode, the code that handles redirect() call is using `location.replace` to navigate to external URL. The problem is that that happens while URL is still set to `/` so it replaces that navigation entry with external URL. Navigating back thus goes back to page visited before `/` which is unexpected.

The solution would be to either allow the URL to change `/redirect` and then trigger `location.replace` or use `location.assign` instead. Former is probably more risky as it would trigger extra routing events and potentially some more code that could possibly fail.

This is only an issue with redirects to external URLs.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8658">#c8658</a>)</em></sub></div>

**Comment by rchl** [comment]
Non-ideal workaround would be to trigger redirect from a short timeout. Not ideal because the `redirect.vue` page will render before redirecting and that looks ugly.

**Comment by rchl** [comment]
reproducible

**Comment by IPRIT** [comment]
Redirect function now is useless in Nuxt.js, because of this bug. Any updates?

**Comment by rchl** [comment]
go away stale bot

**Comment by rrdirnens** [comment]
Have there been any issues opened for this? Is there a fix?

**Comment by TimvdEijnden** [comment]
@rchl  I know this is an old issue but it seems history it not working at all in this example? 

When clicking `Go to next page` I'm also unable to return to index using browser buttons.

**Comment by rchl** [comment]
@TimvdEijnden You need to export to zip from codesandbox.
